### DISCUSSION TITLE
Secman V2

### DISCUSSION THREAD

**Discussion by abdfnx** [scmn-dev/secman#83] [Ideas]
Ok, now secman cli doesn't use [**api.secman.dev**](https://api.secman.dev)

also, when we build secman, this message appeared

![2021-09-24 (10)](https://user-images.githubusercontent.com/64256993/134662715-cb5117be-f905-49f6-8163-73783791b7bb.png)

and secman cli should login to `secman api` not github

> to be honest, I don't like this cli, we have to make secman cli use our api and database

so, I created a small website to see the next secman

[**secman-v2.secman.dev**](https://secman-v2.secman.dev)

# Secman V2

## List Passwords

<p align="center">
   <img src="https://assets.secman.dev/v2/terminal.svg" alt="new secman"  />
</p>

## Authorize

```bash
$ secman auth

E-Mail: yo@secman.dev
MasterPassword: **********
```

## Insert

```bash
$ secman insert -t/--type login

Type: Login
Title: Expo
URL: expo.dev
Username: SecmanX
Password: ********
Extra:
```

## Read

```bash
$ secman read Expo

Title: Expo
URL: expo.dev
Username: SecmanX
Password: ********
Extra:

# show password

$ secman read Expo --show-password

Title: Expo
URL: expo.io
Username: SecmanX
Password: secman_expo_31137
Extra:
```

## Edit

```bash
$ secman edit Expo

What Do You Want To Change ?

1. Title
> 2. URL
3. Username
4. Password
5. Extra

Enter The New URL: expo.dev

# choose

$ secman edit Expo -e/--extra

Enter The New Extra: expo is a powerfull tool
```

## Generate

```bash
$ secman generate
thlnchhmzdwzura

# with options

$ secman generate -l 10 -n -s -c
~-zEQpYXaK

# -l: Length of the generated password
# -n: Include numbers
# -s: Include symbols
# -c: Include capital-letters

# full

$ secman generate --length 16 --numbers --symbols --capital-letters
+Bu-L^uv-/*gQmN
```

## Commands

- **`auth`**
- **`insert`** | **`new`**
- **`read`** | **`show`**
- **`edit`**
- **`generate`** | **`gen`**
- **`delete`** | **`remove`** | **`rm`**
- **`upgrade`** | **`upg`**
- **`uninstall`** | **`un`**
- **`version`**

## Maybe Added

- **`change-master-password`** | **`cms`**
- **`find`** | **`search`**

### I know this maybe hard, but if we did it, we will be better

**Comment by david-tomson** [comment]
ok @abdfnx , I think this will take a long time

but I want to do this 👍

**Reply by Timothee-Cardoso** [reply]
you're right, maybe it will take a months

**Comment by Timothee-Cardoso** [comment]
but I see this secman is good, why we've to change it ?

**Reply by abdfnx** [reply]
I don't like it, we can create a repo for the old secman, just this

**Reply by Timothee-Cardoso** [reply]
no problem

**Comment by abdfnx** [comment]
Now, we should finish all tasks

and start work on **secman v2**

**Comment by abdfnx** [comment]
#84
